### PR TITLE
Spike in calc, lines on diag plots

### PR DESCRIPTION
**Pull request by alanocallaghan**

### CODE REVIEW

File: R/BASiCS_DiagHist.R [hunk 1]
```diff
@@ -5,30 +5,30 @@
...
-#' \code{'sigma2'} and \code{'epsilon'}. Default \code{Parameter = NULL}.
+#' @param Parameter Optional name of a chain parameter to restrict the
+#' histogram; if not supplied, all parameters will be assessed.
+#' Default \code{Parameter = NULL}.
 #' @param Measure Character scalar specifying the diagnostic measure to plot.
-#' Current options are effective sample size and the Geweke diagnostic
-#' criterion.
-#' @param na.rm Logical value indicating whether NA values should be removed
+#' Current options are effective sample size, the Geweke diagnostic
+#' criterion, and the $\hat{R}$ diagnostic, see \code{\link[posterior]{rhat}}.
```
Comment (catavallejos) [inline R/BASiCS_DiagHist.R:13]: Change $\hat{R}$ to \code{R.hat} to avoid warnings in the checks

Comment (catavallejos) [inline R/BASiCS_DiagHist.R:13]:
```suggestion
#' criterion, and the \code{R.hat} diagnostic, see \code{\link[posterior]{rhat}}.
```

File: R/BASiCS_DiagPlot.R [hunk 2]
```diff
@@ -5,54 +5,54 @@
...
-#' \code{'s'}, \code{'nu'}, \code{'theta'}, \code{'beta'},
-#' \code{'sigma2'} and \code{'epsilon'}. Default \code{Parameter = 'mu'}
+#' @param Parameter Name of the parameter to be plotted.
+#' Default \code{Parameter = 'mu'}
 #' @param Measure Character scalar specifying the diagnostic measure to plot.
-#' Current options are effective sample size and the Geweke diagnostic
-#' criterion.
-#' @param x,y Optional MCMC parameter values to be plotted on the x or y axis, 
+#' Current options are effective sample size, the Geweke diagnostic
+#' criterion, and the $\hat{R}$ diagnostic, see \code{\link[posterior]{rhat}}.
```
Comment (catavallejos) [inline R/BASiCS_DiagPlot.R:12]:
```suggestion
#' criterion, and the \code{R.hat} diagnostic, see \code{\link[posterior]{rhat}}.
```

File: R/BASiCS_CalculateSpikeIns.R [hunk 3]
```diff
@@ -0,0 +1,23 @@
+#' Convert concentration in moles per microlitre to molecule counts
+#' @param ERCCConcentration A vector of concentrations, in moles per litre,
+#' of the spike-in genes in the mixture used.
+#' @param DilutionFactor The dilution factor applied to the spike-in mixture.
+#' e.g., 1μl per 50ml would be a 1/50000 \code{DilutionFactor}.
+#' @param VolumePerCell The volume of spike-in mixture added to each well, or to
+#' each cell.
+#' @return The molecule counts per well, or per cell, based on the input
+#' parameters.
+BASiCS_CalculateSpikeIns <- function(
```
Comment (catavallejos) [inline R/BASiCS_CalculateSpikeIns.R:10]: `ERCCConcentration` is an input for this function for not used within it. Moreover `ERCC_mmul` is not defined. 

This is the definition we had in the original workflow:

```
ERCC_mmul <- ERCC_conc$concentration.in. Mix.1.attomoles.ul. * (10ˆ(-18))
```

Are we going to have the ERCC concentration file (`ERCC_conc` in the workflow) pre-loaded in the library?

**Review by catavallejos — CHANGES_REQUESTED** [review]

File: R/BASiCS_CalculateSpikeIns.R (hunk 3 above)
Comment (Alanocallaghan) [reply R/BASiCS_CalculateSpikeIns.R]: I think my idea was we'd pass in the conc vector but this makes more sense

File: R/BASiCS_DiagPlot.R (hunk 2 above)
Comment (Alanocallaghan) [reply R/BASiCS_DiagPlot.R]: It's posterior::rhat rather than R.hat, but good shout

**Review by catavallejos — APPROVED** [review]

**Pull request merged**